### PR TITLE
Add DealSponsored event handler 

### PR DESCRIPTION
**Pull request by gfournieriExec**

### CODE REVIEW

File: src/Modules/IexecPoco.ts [hunk 1]
```diff
@@ -421,3 +423,11 @@ export function handleFaultyContribution(event: FaultyContributionEvent): void {
     workerAccount.score = faultyContributionEvent.score;
     workerAccount.save();
 }
+export function handleDealSponsored(event: DealSponsoredEvent): void {
```
Comment (zguesmi) [inline src/Modules/IexecPoco.ts:426]:
```suggestion

export function handleDealSponsored(event: DealSponsoredEvent): void {
```

File: schema.graphql [hunk 2]
```diff
@@ -436,6 +436,14 @@ type OrdersMatched implements DealEvent @entity {
   deal: Deal!
 }
 
+type DealSponsored implements DealEvent @entity {
+  id: ID!
+  transaction: Transaction!
+  timestamp: BigInt!
+  deal: Deal!
```
Comment (zguesmi) [inline schema.graphql:443]: Why not?
```suggestion
  dealId: Deal!
```

File: test/Modules/IexecPoco.test.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { Address, BigInt, Bytes, ethereum } from '@graphprotocol/graph-ts';
+import { assert, describe, newTypedMockEventWithParams, test } from 'matchstick-as/assembly/index';
+import { DealSponsored } from '../../generated/Core/IexecInterfaceToken';
+import { handleDealSponsored } from '../../src/Modules';
+
+describe('IexecPoco', () => {
+    test('Should handle DealSponsored', () => {
+        // Define mock parameters
+        const dealId = Bytes.fromHexString('0x1234567890abcdef1234567890abcdef12345678');
+        const sponsor = Address.fromString('0xabcdef1234567890abcdef1234567890abcdef12');
```
Comment (zguesmi) [inline test/Modules/IexecPoco.test.ts:13]:
```suggestion
        const sponsor = Address.fromHexString('0xabcdef1234567890abcdef1234567890abcdef12');
```

File: test/Modules/IexecPoco.test.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+import { Address, BigInt, Bytes, ethereum } from '@graphprotocol/graph-ts';
+import { assert, describe, newTypedMockEventWithParams, test } from 'matchstick-as/assembly/index';
+import { DealSponsored } from '../../generated/Core/IexecInterfaceToken';
+import { handleDealSponsored } from '../../src/Modules';
+
+describe('IexecPoco', () => {
+    test('Should handle DealSponsored', () => {
+        // Define mock parameters
+        const dealId = Bytes.fromHexString('0x1234567890abcdef1234567890abcdef12345678');
```
Comment (zguesmi) [inline test/Modules/IexecPoco.test.ts:12]: It should be 32 bytes
```suggestion
        const dealId = Bytes.fromHexString('0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef');
```

**Review by Le-Caignec — COMMENTED** [review]
Just one commentary

File: src/Modules/IexecPoco.ts [hunk 5]
```diff
@@ -421,3 +423,11 @@ export function handleFaultyContribution(event: FaultyContributionEvent): void {
...
+    let dealSponsoredEvent = new DealSponsored(createEventID(event));
+    dealSponsoredEvent.transaction = logTransaction(event).id;
```
Comment (Le-Caignec) [inline src/Modules/IexecPoco.ts:428]: you should load an entity before creating it. It prevent bugs when the graphNode make some re-indexing behavior: 

```suggestion
    let dealSponsored = DealSponsored.load(createEventID(event))
    if (!dealSponsored){
            dealSponsored = new DealSponsored(createEventID(event));
    }
     dealSponsored.transaction = logTransaction(event).id;
```

**Review by zguesmi — APPROVED** [review]

**Review by jeremyjams — APPROVED** [review]
Thanks :)

**Pull request merged**